codegen/smithy-aws-go-codegen/src/main/java/software/amazon/smithy/aws/go/codegen/RestJsonProtocolGenerator.java

```diff
@@ -683,6 +685,7 @@ private void writeErrorShapeDeserializerDelegator(GoWriter writer, Model model,
             writer.openBlock("if errorType == $S {", "}", errorShapeId.getName(), () -> {
                 writer.write("errResult := &$T{}", errorSymbol);
                 writer.write("output := errResult");
+                writer.write("_ = output");
```

Comment:
what is this used for?

Reply:
yeah this is needed because it is possible for the `output` member never to be referenced. 

```go
		if errorType == "FooError" {
			errResult := &types.FooError{}
			output := errResult
			_ = output

			response.StatusCode = 500
			response.Status = http.StatusText(500)

			return out, metadata, errResult
		}
```